Sources/Script/Script.swift

```diff
@@ -145,7 +145,21 @@ public class Script {
             // if the Swift version is less than 5 we are not an ABI safe environment
             guard let swiftVersion = Float(swiftVersion), swiftVersion >= 5 else { return true }
 
-            if let t1 = path.mtime, let t2 = binaryPath.mtime {
+            // compute latest mtime for script and local dependencies
+            var mtimes = [path.mtime]
+            for dep in deps {
+                switch dep.dependencyName {
+                case .local(let path):
+                    for path in path.find() {
```

Comment:
Seems like more work than necessary, to scan every file. But fine as a initial implementation that can be optimized.

mtime of the parent dir should suffice on mac. Linux varies.

Reply:
Hmm, can't reproduce file mtime changes affecting the directory.

```
$ ls -al
total 56
drwxr-xr-x@ 14 josh  staff   448 Apr  1 16:13 .
-rw-r--r--   1 josh  staff  1275 Apr  1 19:19 LICENSE.md
$ touch LICENSE.md 
$ ls -al          
drwxr-xr-x@ 14 josh  staff   448 Apr  1 16:13 .
-rw-r--r--   1 josh  staff  1275 Apr  1 19:21 LICENSE.md
```